apiserver/facades/agent/secretsmanager/secrets.go

```diff
@@ -81,6 +81,47 @@ func (s *SecretsManagerAPI) createSecret(ctx context.Context, arg params.CreateS
 	return md.URL.String(), nil
 }
 
+// UpdateSecrets updates the specified secrets.
+func (s *SecretsManagerAPI) UpdateSecrets(args params.UpdateSecretArgs) (params.StringResults, error) {
+	result := params.StringResults{
+		Results: make([]params.StringResult, len(args.Args)),
+	}
+	ctx := context.Background()
+	for i, arg := range args.Args {
+		ID, err := s.updateSecret(ctx, arg)
+		result.Results[i].Result = ID
+		result.Results[i].Error = apiservererrors.ServerError(err)
+	}
+	return result, nil
+}
+
+func (s *SecretsManagerAPI) updateSecret(ctx context.Context, arg params.UpdateSecretArg) (string, error) {
+	URL, err := coresecrets.ParseURL(arg.URL)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	if URL.Attribute != "" {
+		return "", errors.NotSupportedf("updating a single secret attribute %q", URL.Attribute)
+	}
+	if URL.Revision > 0 {
+		return "", errors.NotSupportedf("updating secret revision %d", URL.Revision)
+	}
+	if arg.RotateInterval < 0 && len(arg.Data) == 0 {
+		return "", errors.New("either rotate interval or data must be specified")
+	}
+	URL.ControllerUUID = s.controllerUUID
```

Comment:
do we wanna reject if the the URL does provide a wrong `ControllerUUID` and/or `ModelUUID`?

Reply:
yeah, done